### PR TITLE
Wallet List Scene Loading Progress Indicator / Bar

### PR DESCRIPTION
**Pull request by kylanhurt**
The purpose of this task is to create a "dropdown" on the Wallet List that shows up initially when the scene loads, and gives the user an update as to what % of their wallets have loaded. The component can also be hidden simply by pressing on it. Otherwise it disappears shortly after reaching 100%

Asana Task: https://app.asana.com/0/361770107085503/556823215362117/f

### CODE REVIEW

File: src/modules/Login/action.js [hunk 1]
```diff
@@ -99,6 +100,7 @@ export const initializeAccount = (account: EdgeAccount, touchIdInfo: Object) =>
       accountInitObject.currencyCode = currencyCode
     }
     const activeWalletIds = account.activeWalletIds
+    dispatch(insertWalletIdsForProgress(activeWalletIds))
```
Comment (wolverineks) [inline src/modules/Login/action.js]: can you grab all the `activeWalletIds` from the login payload on line 163?

File: src/modules/ReduxTypes.js [hunk 2]
```diff
@@ -160,7 +160,8 @@ export type State = {
       selectedWalletId: string,
       selectedCurrencyCode: string,
       addTokenPending: boolean,
-      manageTokensPending: boolean
+      manageTokensPending: boolean,
+      walletLoadingProgress: Object
```
Comment (wolverineks) [inline src/modules/ReduxTypes.js]: `type: { [walletId: string]: number }`

File: src/modules/UI/Wallets/action.js [hunk 3]
```diff
@@ -39,6 +39,8 @@ export const EDIT_CUSTOM_TOKEN_FAILURE = 'EDIT_CUSTOM_TOKEN_FAILURE'
 export const UPDATE_EXISTING_TOKEN_SUCCESS = 'UPDATE_EXISTING_TOKEN_SUCCESS'
 export const OVERWRITE_THEN_DELETE_TOKEN_SUCCESS = 'OVERWRITE_THEN_DELETE_TOKEN_SUCCESS'
 export const ADD_NEW_TOKEN_THEN_DELETE_OLD_SUCCESS = 'ADD_NEW_TOKEN_THEN_DELETE_OLD_SUCCESS'
+export const UPDATE_WALLET_LOADING_PROGRESS = 'UPDATE_WALLET_LOADING_PROGRESS'
+export const INSERT_WALLET_IDS_FOR_PROGRESS = 'INSERT_WALLET_IDS_FOR_PROGRESS'
```
Comment (wolverineks) [inline src/modules/UI/Wallets/action.js]: if you can use the dispatch payload at the end of the login action, then this is not needed

File: src/modules/UI/Wallets/action.js [hunk 4]
```diff
@@ -436,6 +438,30 @@ export function addNewTokenThenDeleteOldSuccess (data: any) {
...
+    walletId,
+    addressLoadingProgress
+  }
+  return {
+    type: UPDATE_WALLET_LOADING_PROGRESS,
+    data
+  }
+}
+
+export function insertWalletIdsForProgress (activeWalletIds: Array<string>) {
```
Comment (wolverineks) [inline src/modules/UI/Wallets/action.js]: if you use the payload from the dispatch at the end of the login action, then this is not needed
additionally, try to keep data munging in the reducers

File: src/modules/UI/Wallets/reducer.js [hunk 5]
```diff
@@ -182,6 +182,22 @@ export const walletEnabledTokens = (state: any = {}, action: Action) => {
   return state
 }
 
+export const walletLoadingProgress = (state: Object = {}, action: Action) => {
+  if (!action.data) return state
+  const {type, data} = action
+  switch (type) {
+    case ACTION.INSERT_WALLET_IDS_FOR_PROGRESS:
```
Comment (wolverineks) [inline src/modules/UI/Wallets/reducer.js]: if you can grab the payload from the login dispatch...
`case Constants.AccountInitComplete:`

Comment (wolverineks) [reply src/modules/UI/Wallets/reducer.js]: additionally, do any data munging here

File: src/modules/UI/components/SimpleDropdown/SimpleDropdown.ui.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+// @flow
+
+import React, { Component } from 'react'
+import { TouchableWithoutFeedback } from 'react-native'
+import styles from './style.js'
+import * as Animatable from 'react-native-animatable'
+
+export type Props = {
+  height?: number,
+  children: any,
```
Comment (wolverineks) [inline src/modules/UI/components/SimpleDropdown/SimpleDropdown.ui.js]: `type Node`

File: src/modules/UI/components/SimpleDropdown/SimpleDropdown.ui.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  onPress: Function,
```
Comment (wolverineks) [inline src/modules/UI/components/SimpleDropdown/SimpleDropdown.ui.js]: `type: () => void`

File: src/modules/UI/components/SimpleDropdown/SimpleDropdown.ui.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  containerStyle: Object
```
Comment (wolverineks) [inline src/modules/UI/components/SimpleDropdown/SimpleDropdown.ui.js]: `type: StyleSheet.Styles`

File: src/modules/UI/components/SimpleDropdown/SimpleDropdown.ui.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export type State = {
+
+}
+
+export default class WalletListProgressDropdown extends Component<Props> {
+  render () {
+    return (
+      <Animatable.View animation={'fadeIn'} duration={6000} /* slideInDown */ style={[styles.container, this.props.containerStyle, {height: this.props.height}]}>
```
Comment (wolverineks) [inline src/modules/UI/components/SimpleDropdown/SimpleDropdown.ui.js]: please remove unused code

File: src/modules/UI/scenes/WalletList/WalletList.ui.js [hunk 10]
```diff
@@ -56,7 +58,8 @@ type Props = {
   updateActiveWalletsOrder: (Array<string>) => void,
   walletRowOption: (walletId: string, option: string, archived: boolean) => void,
   disableOtp: () => void,
-  keepOtp: () => void
+  keepOtp: () => void,
+  progressPercentage: Number
```
Comment (wolverineks) [inline src/modules/UI/scenes/WalletList/WalletList.ui.js]: `type: number`

File: src/modules/UI/scenes/WalletList/WalletListConnector.js [hunk 11]
```diff
@@ -25,6 +25,23 @@ const mapStateToProps = (state: State) => {
   const dimensions = state.ui.scenes.dimensions
   const customTokens = state.ui.settings.customTokens
   const otpResetPending = SETTINGS_SELECTORS.getOtpResetPending(state)
+  const walletsForProgress = state.ui.wallets.walletLoadingProgress
```
Comment (wolverineks) [inline src/modules/UI/scenes/WalletList/WalletListConnector.js]: can you put all this in a selector?
```
const getWalletLoadingPercent = (loadingWallets) => {
  return // ...do the data munging...
}

const mapStateToProps = (state) {
  walletLoadingPercent: getWalletLoadingPercent(state.ui.wallets.loadingWallets)
}

**Comment by kylanhurt** [comment]
@wolverineks I made the changes requested. I was *unable* to move the initial activeWalletId's data dispatch into the ACCOUNT_INIT_COMPLETE dispatch, however, since there is asynchronous code in that routine that can cause initial wallet progress callbacks (onAddressesChecked) to arrive before the reducer has the list of activeWalletIds. For that reason I kept the initial dispatch right when it the routing first finds out about the list (of activeWalletIds). I did incorporate the rest of your comments, though.

**Pull request closed without merging**